modules/launcher/src/main/java/org/mule/runtime/module/launcher/ToolingServiceAware.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.launcher;
+
+import org.mule.runtime.module.tooling.api.ToolingService;
+
+/**
+ * Enables {@link ToolingServiceAware} injection.
+ *
+ * @since 4.0
+ */
+public interface ToolingServiceAware
```

Comment:
ToolingService seems like a too broad term, very likely to quickly grow into a cat bag. I would create more specific interfaces, clearly separating concerns. Otherwise ToolingService will know about connectivity testing, dataSense, types, etc, etc